org-formation/075-security-hub/security-hub-suppress-infra.yaml

```diff
@@ -234,22 +231,134 @@ Resources:
+          """
+          Uses the Lambda Powertools BatchProcessor in order to build the findings id list.
+          Lambda powertools is used to prevent successfully processed messages being returned to SQS
+          https://awslabs.github.io/aws-lambda-powertools-python/latest/utilities/batch/
+          """
+          @logger.inject_lambda_context(log_event=True)
+          def handler(event, context):
+              records = event["Records"]
+              logger.info(f"records {records}")
+
+              processor = BatchProcessor(event_type=EventType.SQS)
+
+              rh = RecordsHandler()
+              with processor(records, rh.record_handler) as proc:
+                  proc.process()
+
+              finding_identifiers = rh.get_finding_identifiers()
+              logger.info(f"finding_identifiers {finding_identifiers}")
+
+              try:
+                  response = sh_client.batch_update_findings(
+                      FindingIdentifiers=finding_identifiers,
+                      Workflow={"Status": "SUPPRESSED"},
+                  )
+                  for processed_findings in response["ProcessedFindings"]:
+                      logger.info(
+                          f"processed and suppressed id {processed_findings['Id']} productarn {processed_findings['ProductArn']}"
+                      )
+
+                  for unprocessed_findings in response["UnprocessedFindings"]:
+                      logger.error(
+                          f"unprocessed finding id {unprocessed_findings['FindingIdentifier']['Id']} productarn {unprocessed_findings['FindingIdentifier']['ProductArn']} error code {unprocessed_findings['ErrorCode']} error message {unprocessed_findings['ErrorMessage']}"
+                      )
+
+              except boto3exceptions.ClientError as error:
+                  logger.exception("client error")
+                  raise ConnectionError(f"Client error invoking batch update findings {error}")
+              except boto3exceptions.ParamValidationError as error:
+                  logger.exception("invalid parameters")
+                  raise ValueError(f"The parameters you provided are incorrect: {error}")
+
+              return {"statusCode": 200}
+      Role: !GetAtt SecurityHubBatchUpdateSuppressionRole.Arn
+      FunctionName: suppress-security-hub-findings
+      Description: Suppression lambda for Security Hub implements batch update findings
+      Handler: index.handler
+      Layers:
+      - Fn::GetAtt:
+        - AWSLambdaPowertoolsApplication
+        - Outputs.LayerVersionArn
+      Runtime: python3.8
+      Timeout: 120
+  BatchupdateEventInvokeConfig:
+    Type: AWS::Lambda::EventInvokeConfig
+    Properties:
+      FunctionName: suppress-security-hub-findings
+      Qualifier: "$LATEST"
+      MaximumRetryAttempts: 2
+  BatchUpdateLambdaFctEventSourceEventSourceMapping:
+    Type: AWS::Lambda::EventSourceMapping
+    Properties:
+      FunctionName: suppress-security-hub-findings
```

Comment:
less places with hard coded names would be better therefore i would consider doing a `!Ref BatchupdateLambdaFct` here